src/gtfs/write.rs

```diff
@@ -146,9 +147,55 @@ pub fn write_trips(
     Ok(())
 }
 
+fn ntfs_geometry_to_gtfs_shapes(g: &objects::Geometry) -> Vec<Shape> {
```

Comment:
You can try to return `impl Iterator<Item=Shape>`, but it will be a bit tricky to have the same return type in the 2 cases.

A solution would be to extract the `&LineString` with using an empty one in the other case, and then `points().enumerate()`...